quickcheck-dynamic-iosim/src/Test/QuickCheck/StateModel/IOSim.hs

```diff
@@ -3,24 +3,21 @@
 module Test.QuickCheck.StateModel.IOSim where
 
 import Control.Concurrent
+import Control.Concurrent.Class.MonadSTM.TMVar qualified as IOClass
+import Control.Concurrent.Class.MonadSTM.TVar qualified as IOClass
 import Control.Concurrent.STM
 import Control.Monad.Class.MonadFork qualified as IOClass
-import Control.Monad.Class.MonadSTM qualified as IOClass
 import Control.Monad.IOSim
 
--- TODO: when we've updated the dependency on io-sim
--- import Control.Monad.Class.MonadMVar qualified as IOClass
-
 import Test.QuickCheck.StateModel
 
 type family RealizeIOSim s a where
   RealizeIOSim s ThreadId = IOClass.ThreadId (IOSim s)
   RealizeIOSim s (TVar a) = IOClass.TVar (IOSim s) a
--- TODO: when we've updated the dependency on io-sim
--- RealizeIOSim s (MVar a) = IOClass.MVar (IOSim s) a
+  RealizeIOSim s (MVar a) = IOClass.TMVar (IOSim s) a
 -- TODO: unfortunately no poly-kinded recursion for type families
 -- so we can't do something like :'(
--- RealizeIOSim s (f a)    = (RealizeIOSim f) (RealizeIOSim s a)
+-- RealizeIOSim s (f a)  = (RealizeIOSim f) (RealizeIOSim s a)
```

Comment:
suggestion: Remove commented code

Reply:
Yeah I don't know what to do about this. I want to implement something like this but I'm not sure we have a solution?

Reply:
YAGNI :)

Reply:
que?

Reply:
You Ain't Gonna Need It :-D

Reply:
https://martinfowler.com/bliki/Yagni.html